File: CPP/Program/Garage.cpp
#pragma once

#include <bits/stdc++.h>
#include <string>
#include <vector>
#include "Vehicle.cpp"

using namespace std;

class Garage {
private:
    string nama_garasi;
    int luas_garasi;
    string daftar_kendaraan;
    list<Vehicle*> kendaraan;  // Change this to a list of Vehicle pointers

public:
    Garage(){

    }

    Garage(string nama_garasi, int luas_garasi, string daftar_kendaraan, list<Vehicle*> kendaraan){
        this->nama_garasi = nama_garasi;
        this->luas_garasi = luas_garasi;
        this->daftar_kendaraan = daftar_kendaraan;
        this->kendaraan = kendaraan;
    }

    // Getter
    string getNamaGarasi() { 
        return nama_garasi; 
    }

    int getLuasGarasi() { 
        return luas_garasi; 
    }
    
    string getDaftarKendaraan() { 
        return daftar_kendaraan; 
    }

    const list<Vehicle*>& getKendaraan() const {  // Return a const reference to the list of Vehicle pointers
        return kendaraan; 
    }

    // Setter
    void setNamaGarasi(string nama_garasi) { 
        this->nama_garasi = nama_garasi; 
    }
    
    void setLuasGarasi(int luas_garasi) { 
        this->luas_garasi = luas_garasi;
    }

    void setDaftarKendaraan(string daftar_kendaraan) { 
        this->daftar_kendaraan = daftar_kendaraan; 
    }

    void setKendaraan(const list<Vehicle*>& kendaraan) { 
        this->kendaraan = kendaraan;
    }

    void DisplayGarage() const {
        cout << "Nama Garasi: " << nama_garasi << endl;
        cout << "Luas Garasi: " << luas_garasi << "meter persegi" << endl;
        cout << "Daftar Kendaraan: " << daftar_kendaraan << endl;

        // Display details of each vehicle in the garage
        for (const auto& vehicle : kendaraan) {
            vehicle->DisplayDetails();
        }
    }

    ~Garage(){
        
    }
};

File: Python/Program/Garage.py
from Car import Car
from Motorcycle import Motorcycle
from Vehicle import Vehicle

class Garage:
    __namaGarasi = ""
    __luasGarasi = ""
    __daftarKendaraan = ""

    def __init__(self, namaGarasi = "", luasGarasi = "", daftarKendaraan = "", car = [], motorcycle = []):
        self.__namaGarasi = namaGarasi
        self.__luasGarasi = luasGarasi
        self.__daftarKendaraan = daftarKendaraan
        self.__car = car
        self.__motorcycle = motorcycle

    def getNamaGarasi(self):
        return self.__namaGarasi
    
    def setNamaGarasi(self, namaGarasi):
        self.__namaGarasi = namaGarasi

    def getLuasGarasi(self):
        return self.__luasGarasi

    def setLuasGarasi(self, luasGarasi):
        self.__luasGarasi = luasGarasi

    def getDaftarKendaraan(self):
        return self.__daftarKendaraan
    
    def setDaftarKendaraan(self, daftarKendaraan):
        self.__daftarKendaraan = daftarKendaraan

    def getCar(self):
        return self.__car
    
    def setCar(self, car):
        self.__car = car

    def getMotorcycle(self):
        return self.__motorcycle
    
    def setMotorcycle(self, motorcycle):
        self.__motorcycle = motorcycle
        